### PR TITLE
feat: added upload-api methods

### PR DESCRIPTION
**Pull request by edge33**

### CODE REVIEW

**Review by alewin — COMMENTED** [review]
why aren't you using `paths` types from `schema.ts`?

**Comment by SamSalvatico** [comment]
> why aren't you using `paths` types from `schema.ts`?

@alewin is using that, he just removed it from the biome config so it doesn't format it, given that is autogenerated

**Comment by alewin** [comment]
> > why aren't you using `paths` types from `schema.ts`?
> 
> I am using the types from the schema, did I miss something? 👀

in order to avoid refactoring the SDK in case there are changes to the API, we can use the `paths` types in each SDK methods
I'm taking `getFilesMetadata` as an example, which can be something like this
```ts
  getFilesMetadata(params: paths["/api/v1/metadata"]["get"]["parameters"]["query"]) {
```

**Comment by SamSalvatico** [comment]
> > > why aren't you using `paths` types from `schema.ts`?
> > 
> > 
> > I am using the types from the schema, did I miss something? 👀
> 
> in order to avoid refactoring the SDK in case there are changes to the API, we can use the `paths` types in each SDK methods I'm taking `getFilesMetadata` as an example, which can be something like this
> 
> ```ts
>   getFilesMetadata(params: paths["/api/v1/metadata"]["get"]["parameters"]["query"]) {
> ```

@alewin can be useful sometimes, we are using it for some methods
But we have to remember that the SDKs will be public, so it would be better to be strict as possible with params, even if it could be redundant

**Review by SamSalvatico — APPROVED** [review]

File: src/client/clients/upload/index.ts [hunk 1]
```diff
@@ -23,17 +22,107 @@ class Upload extends BaseClient<paths> {
...
+  getSharedFilesForUser(userId: string) {
+    this.client
+      .GET("/api/v1/metadata/", {
+        params: { query: { userId } },
+      })
+      .then(
+        (response) => this.formatResponse(response),
+        (reason) => this.formatError(reason),
+      );
+  }
```
Comment (alewin) [inline src/client/clients/upload/index.ts:83]:
```suggestion
  getSharedFilesForUser(userId: string) {
    return this.client
      .GET("/api/v1/metadata/", {
        params: { query: { userId } },
      })
      .then(
        (response) => this.formatResponse(response),
        (reason) => this.formatError(reason),
      );
  }
```

File: src/client/clients/upload/index.ts [hunk 2]
```diff
@@ -23,17 +22,107 @@ class Upload extends BaseClient<paths> {
...
+  getFileMetadata(id: string) {
+    this.client
+      .GET("/api/v1/metadata/{id}", {
+        params: { path: { id } },
+      })
+      .then(
+        (response) => this.formatResponse(response),
+        (reason) => this.formatError(reason),
+      );
+  }
```
Comment (alewin) [inline src/client/clients/upload/index.ts:94]:
```suggestion
  getFileMetadata(id: string) {
    return this.client
      .GET("/api/v1/metadata/{id}", {
        params: { path: { id } },
      })
      .then(
        (response) => this.formatResponse(response),
        (reason) => this.formatError(reason),
      );
  }
```

File: src/client/clients/upload/index.ts [hunk 3]
```diff
@@ -23,17 +22,107 @@ class Upload extends BaseClient<paths> {
...
+          formData.set("expirationDate", pasrsed.expirationDate);
+        }
+        formData.set("file", pasrsed.file);
+        return formData;
       },
-    };
+    });
 
-    this.client.use(authMiddleware);
+    return { error };
```
Comment (alewin) [inline src/client/clients/upload/index.ts:125]: what about `data`?

**Comment by alewin** [comment]
> But we have to remember that the SDKs will be public, so it would be better to be strict as possible with params, even if it could be redundant

Yes, we can be more strict, but we must also maintain a type-safe SDK according to the API specification
If `getFilesMetadata` changes the name of the `organisationId` parameter to `organisation_Id` in the SDK you will not get any warnings, or even if a new parameter is added in the body of a request, we have to retouch the SDK each time
if you want, we can merge for now, but this is an issue we have to take into account, also to make the other clients of the SDK uniform with a standard approch/pattern

File: src/client/clients/upload/index.ts (hunk 3 above)
Comment (edge33) [reply src/client/clients/upload/index.ts:125]: we do not return data in this case, we just create a resource

File: src/client/clients/upload/index.ts (hunk 1 above)
Comment (edge33) [reply src/client/clients/upload/index.ts:83]: good catch thanks

**Comment by edge33** [comment]
> > But we have to remember that the SDKs will be public, so it would be better to be strict as possible with params, even if it could be redundant
> 
> Yes, we can be more strict, but we must also maintain a type-safe SDK according to the API specification If `getFilesMetadata` changes the name of the `organisationId` parameter to `organisation_Id` in the SDK you will not get any warnings, or even if a new parameter is added in the body of a request, we have to retouch the SDK each time if you want, we can merge for now, but this is an issue we have to take into account, also to make the other clients of the SDK uniform with a standard approch/pattern

this is indeedd a great suggestion, I will make sure to check if for future changes, to the sdk, I would merge for now, in order to proceed testing on LE

**Review by SamSalvatico — APPROVED** [review]

**Review by alewin — APPROVED** [review]

**Pull request merged**